### PR TITLE
Removal route pfw ndessert

### PR DESCRIPTION
**Pull request by cc6565**

### CODE REVIEW

**Comment by dawagner** [comment]
Wow. The serial patcher strikes again :)

File: Android.mk [hunk 1]
```diff
@@ -19,6 +19,21 @@
...
 
 LOCAL_PATH := $(call my-dir)
+HAL_COMMON_CFLAGS :=
+
+USE_ALSA_LIB := 0
+ifeq ($(USE_ALSA_LIB), 1)
+HAL_COMMON_CFLAGS := $(HAL_COMMON_CFLAGS) -DUSE_ALSA_LIB
+endif
+
+HAVE_BOOST := 0
```
Comment (plbossart) [inline Android.mk]: yuck

File: Android.mk [hunk 2]
```diff
@@ -19,6 +19,21 @@
 ifeq ($(INTEL_AUDIO_HAL),audio_pfw)
 
 LOCAL_PATH := $(call my-dir)
+HAL_COMMON_CFLAGS :=
+
+USE_ALSA_LIB := 0
```
Comment (plbossart) [inline Android.mk]: ??

File: audio_route_manager/StreamRouteConfig.cpp [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    struct mixer_ctl *ctl;
+    unsigned int channelMaskSize;
+    int channelCount = 0;
+
+    int cardIndex = AudioUtils::getCardIndexByName(cardName.c_str());
+    if (cardIndex < 0) {
+        Log::Error() << __FUNCTION__ << ": Failed to get Card Name index " << cardIndex;
+        return android::BAD_VALUE;
+    }
+    mixer = mixer_open(cardIndex);
```
Comment (plbossart) [inline audio_route_manager/StreamRouteConfig.cpp]: i don't get what this is doing and why we need to get mixer control values from ALSA?

File: audio_route_manager/config/audio_criteria.xml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    <criterion name="VoIPBandType" type="BandType" default="NB" parameter="voip_band_type"/>
+    <criterion name="ImsBandType" type="BandType" default="Unknown" parameter="cme_band"
+               mapping="nb:NB,wb:WB,swb:SuperWB"/>
+    <criterion name="ContextAwareness" type="BooleanType" default="False" parameter="context_awareness_status"
+               mapping="off:False,on:True"/>
+    <criterion name="FmEnabled" type="BooleanType" default="False" parameter="route-fm"
+               mapping="off:False,speaker:True,headset:True"/>
+    <criterion name="AlwaysListening" type="BooleanType" default="False" parameter="vtsv_route"
+               mapping="off:False,on:True"/>
+    <criterion name="LpalDevice" type="InputDeviceType" default="None" parameter="lpal_device"/>
```
Comment (plbossart) [inline audio_route_manager/config/audio_criteria.xml]: should we clean this up? This looks like the union of all parameters ever supported

File: audio_route_manager/config/readme.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    -list of audio parameter-framework criteria
+    -list of audio parameter-framework rogue parameters
+
+For more documentation on the PFW, please refer to the github link:
+https://github.com/01org/parameter-framework
+
+
+@TODO: Merge the audio_policy_configuration.xml and route_manager_configuration.xml by enhancing
+       the mix ports information.
+
```
Comment (plbossart) [inline audio_route_manager/config/readme.md]: to be clear, this is not something we can do on our own, it has dependencies on AOSP changes, right?

File: audio_route_manager/config/route_manager_configuration.xml [hunk 6]
```diff
@@ -0,0 +1,204 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<routeConfiguration version="1.0" xmlns:xi="http://www.w3.org/2001/XInclude">
+
+    <xi:include href="audio_rogue_parameters.xml"/>
+    <xi:include href="audio_criterion_types.xml"/>
+    <xi:include href="audio_criteria.xml"/>
+
+    <mixPorts>
+        <mixPort name="Media" role="source" card="broxtongpmrb" device="0"
```
Comment (plbossart) [inline audio_route_manager/config/route_manager_configuration.xml]: I am not clear why we have a reference to the MRB here?
Or put differently, this should be board dependent and placed in device/intel/whatever

File: audio_route_manager/AudioCapabilities.cpp [hunk 7]
```diff
@@ -85,6 +86,28 @@ void AudioCapability::reset()
     }
 }
 
+android::status_t AudioCapability::dump(const int fd, int spaces, bool isOut) const
+{
+    const size_t SIZE = 512;
```
Comment (sguiriec) [inline audio_route_manager/AudioCapabilities.cpp]: How size if decided? sometimes 512 and sometimes 256.

File: Android.mk [hunk 8]
```diff
@@ -1,6 +1,6 @@
 #
 #
-# Copyright (C) Intel 2013-2016
```
Comment (sguiriec) [inline Android.mk]: Why?

File: parameter_mgr_helper/Parameter.cpp [hunk 9]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2014-2015 Intel Corporation
```
Comment (sguiriec) [inline parameter_mgr_helper/Parameter.cpp]: Why?

File: parameter_mgr_helper/Parameter.cpp [hunk 10]
```diff
@@ -1,5 +1,5 @@
...
+ * Copyright (C) 2014-2017 Intel Corporation
```
Comment (sguiriec) [inline parameter_mgr_helper/Parameter.cpp]: Why?

File: parameter_mgr_helper/includes/Criterion.hpp [hunk 11]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2013-2015 Intel Corporation
```
Comment (sguiriec) [inline parameter_mgr_helper/includes/Criterion.hpp]: Why?

File: parameter_mgr_helper/includes/Criterion.hpp [hunk 12]
```diff
@@ -1,5 +1,5 @@
...
+ * Copyright (C) 2013-2017 Intel Corporation
```
Comment (sguiriec) [inline parameter_mgr_helper/includes/Criterion.hpp]: Why?

File: utilities/Android.mk [hunk 13]
```diff
@@ -1,6 +1,6 @@
 #
 #
-# Copyright (C) Intel 2015-2016
```
Comment (sguiriec) [inline utilities/Android.mk]: Why?

File: utilities/parameter/test/KeyValuePairsTest.hpp [hunk 14]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2014-2015 Intel Corporation
```
Comment (sguiriec) [inline utilities/parameter/test/KeyValuePairsTest.hpp]: Why?

File: utilities/typeconverter/Android.mk [hunk 15]
```diff
@@ -1,6 +1,6 @@
 #
 #
-# Copyright (C) Intel 2013-2016
+# Copyright (C) Intel 2017
```
Comment (sguiriec) [inline utilities/typeconverter/Android.mk]: Why?

File: Android.mk (hunk 2 above)
Comment (cc6565) [inline Android.mk]: HAL_COMMON_CFLAGS -> good practice to factorize flags amount HAL.
USE_ALSA_LIB -> for SmartX, but i did not want to remove all the work done to support it, so use a flag to remove all dependencies on ALSA Lib

File: audio_route_manager/StreamRouteConfig.cpp (hunk 3 above)
Comment (cc6565) [inline audio_route_manager/StreamRouteConfig.cpp]: We are populating here all the supported channel mask from a mixer control (indirectly read from EDID).

File: audio_route_manager/config/audio_criteria.xml (hunk 4 above)
Comment (cc6565) [inline audio_route_manager/config/audio_criteria.xml]: Mmmh you re right, it is the all-in-wonder criteria supported. I will remove some of them.

File: Android.mk (hunk 1 above)
Comment (cc6565) [inline Android.mk]: HAVE_BOOST -> it is a pity we cannot use boost for string tokenization for example... :-(

File: audio_route_manager/config/readme.md (hunk 5 above)
Comment (cc6565) [inline audio_route_manager/config/readme.md]: As the configuration file is provided by the vendor, we do not have dependencies on AOSP as far as enhanced information found in XML does not disturb the policy deserialization. This is the reason why I try to be the closest as possible from the syntax of audio_policy_configuration.xml

File: audio_route_manager/config/route_manager_configuration.xml (hunk 6 above)
Comment (cc6565) [inline audio_route_manager/config/route_manager_configuration.xml]: it is just provided as an example. This example will be located in system/etc, and vendor files in vendor/etc. Audio HAL will first check for vendor files availability before taking system files.

File: parameter_mgr_helper/Parameter.cpp (hunk 9 above)
Comment (cc6565) [inline parameter_mgr_helper/Parameter.cpp]: this file has been changed, select view all to see these changes

File: Android.mk (hunk 8 above)
Comment (cc6565) [inline Android.mk]: this file has been changed, select view all to see these changes

File: utilities/Android.mk (hunk 13 above)
Comment (cc6565) [inline utilities/Android.mk]: done

File: Android.mk (hunk 2 above)
Comment (plbossart) [inline Android.mk]: It's fine to keep the conditional part, I was more asking if it makes sense to set USE_ALSA_LIB in this makefile. We have other flags in BoardConfig.mk

BOARD_USES_ALSA_AUDIO := true
BOARD_USES_TINY_ALSA_AUDIO := true
BOARD_USES_GENERIC_AUDIO ?= false

so maybe we can clean this up and avoid this new variable at the HAL level.

File: Android.mk (hunk 1 above)
Comment (plbossart) [inline Android.mk]: same comment, if boost is selected for SmartX then do we need to have this new variable here?

File: audio_route_manager/config/readme.md (hunk 5 above)
Comment (plbossart) [inline audio_route_manager/config/readme.md]: I didn't get your point.
Are you saying:
1. we will use a single file but add more information wrapped by new tags in audio_policy_configuration.xml which will be ignored by the policy
2. to use a single file we will need the audio policy to be modified (e.g. mix port definitions)

File: audio_route_manager/config/route_manager_configuration.xml (hunk 6 above)
Comment (plbossart) [inline audio_route_manager/config/route_manager_configuration.xml]: ok makes sense. I wanted to make sure we won't have conflicts with this file between platforms.

File: Android.mk (hunk 2 above)
Comment (cc6565) [inline Android.mk]: the purpose here is to support both in fact (ALSA for SmartX based device, tiny also for real hw device). Not sure it is a real use case however...

File: Android.mk (hunk 1 above)
Comment (cc6565) [inline Android.mk]: boost is also used by AudioHAL, it is so usefull ;-)

**Pull request merged**